### PR TITLE
feat: add uber JAR capabilities

### LINKED ISSUE #24: Add uber JAR-building capabilities
__What should we do?__

- [ ] Add the `maven-shade-plugin` to the POM
- [ ] Document in README how to build the über JAR

__Why should we do it?__

Lower the barrier of providing uber JARs of Connectors, so they can be easily used with the [Connector Runtime](https://github.com/camunda/connector-sdk/tree/main/runtime#automatic-connector-discovery).

### PR DESCRIPTION
**Pull request by tmetzke**
## Description

- Adds uber JAR capabilities.
- Adds SLF4j implementation for unit tests to provide proper logging output.

## Related issues

closes #24

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -108,13 +116,67 @@
...
+              <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
+              <!-- This is needed to not repeat licenses in the META-INF directory -->
+              <transformer implementation="org.apache.maven.plugins.shade.resource.ApacheLicenseResourceTransformer"/>
+              <!-- This is needed to merge existing NOTICE files and keep them downstream -->
+              <transformer implementation="org.apache.maven.plugins.shade.resource.ApacheNoticeResourceTransformer">
+                <addHeader>false</addHeader>
+              </transformer>
+            </transformers>
+            <!-- include relocation to shade dependencies inside your JAR -->
+            <!--<relocations>
```
Comment (igpetrov) [inline pom.xml:150]: Is this an example or why is it commented?

**Review by igpetrov — APPROVED** [review]
Thank you.

File: pom.xml (hunk 1 above)
Comment (tmetzke) [reply pom.xml:150]: Yes, exactly. We can probably highlight that better here 👍

File: pom.xml [hunk 2]
```diff
@@ -108,13 +116,67 @@
...
+              <!-- This is needed if you have dependencies that use Service Loader. Most Google Cloud client libraries does. -->
+              <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
+              <!-- This is needed to not repeat licenses in the META-INF directory -->
+              <transformer implementation="org.apache.maven.plugins.shade.resource.ApacheLicenseResourceTransformer"/>
+              <!-- This is needed to merge existing NOTICE files and keep them downstream -->
+              <transformer implementation="org.apache.maven.plugins.shade.resource.ApacheNoticeResourceTransformer">
+                <addHeader>false</addHeader>
+              </transformer>
+            </transformers>
+            <!-- include relocation to shade dependencies inside your JAR -->
```
Comment (tmetzke) [inline pom.xml:149]:
```suggestion
            <!-- Uncomment and adjust to use relocations inside your JAR -->
```

**Pull request merged**